src/components/certificate-table/CertificatesTable.tsx

```diff
@@ -2,48 +2,97 @@ import './certificate-table.css';
 import { useCertificates } from '../../context/CertificateContext';
 import Table from '../table/Table';
 import Button from '../button/Button';
-import { Link } from 'react-router-dom';
-import { Certificate } from '../../../types/types';
+import { Link, useNavigate } from 'react-router-dom';
+import { Certificate as BaseCertificate } from '../../../types/types';
+import CogIcon from '../../../public/assets/cog.svg';
+import { useEffect, useRef, useState } from 'react';
 
 interface Column<T> {
   Header: string;
   accessor: keyof T;
+  render?: (row: T, rowIndex: number) => React.ReactNode;
 }
 
-const columns: Column<Certificate>[] = [
-  { Header: 'Supplier', accessor: 'supplier' },
-  { Header: 'Certificate type', accessor: 'certificateType' },
-  { Header: 'Valid from', accessor: 'validFrom' },
-  { Header: 'Valid to', accessor: 'validTo' },
-];
-
-/**
- * Example - content wrapper for the certificates table
- * Description - Component which has the certificates table
- * and create certificate button
- * @returns JSX Element
- */
-
-function CertificatesTable(): JSX.Element {
+interface Certificate extends BaseCertificate {
+  actions: React.ReactNode;
+}
+
+const CertificatesTable: React.FC = () => {
   const { certificates } = useCertificates();
+  const navigate = useNavigate();
+  const [dropdownVisible, setDropdownVisible] = useState<number | null>(null);
+  const dropdownRef = useRef<HTMLDivElement | null>(null)
+
+  const handleEdit = (id: number) => {
+    navigate(`/certificates/edit/${id}`);
+  };
+
+  const handleDropdownToggle = (index: number) => {
+    setDropdownVisible(dropdownVisible === index ? null : index);
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+      setDropdownVisible(null);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
```

Comment:
### Feedback:
- Transform code for `handleClickOutside` into a reusable custom hook called useClickOutside. Here's how you can do it:
```JSX
import { useEffect, useRef, RefObject } from 'react';

function useClickOutside<T extends HTMLElement = HTMLElement>(
  handler: () => void
): RefObject<T> {
  const ref = useRef<T>(null);

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        handler();
      }
    };

    document.addEventListener('mousedown', handleClickOutside);

    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [handler]);

  return ref;
}
```
### Benefits
- Improved reusability across different parts of the application
- Better separation of concerns